tests/integration/run_tests.py

```diff
@@ -925,6 +926,41 @@ async def test_getting_paginated_folders(sess):
         assert resp.status == 400
 
 
+async def test_getting_folders_filtered_by_name(sess):
+    """Check that /folders returns folders filtered by name.
+
+    :param sess: HTTP session in which request call is made
+    """
+    names = [" filter new ", "_filter_", "-filter-", "_extra-", "_2021special_"]
+    folders = []
+    for name in names:
+        folder_data = {"name": f"Test{name}name", "description": "Test filtering name"}
+        folders.append(await post_folder(sess, folder_data))
+
+    async with sess.get(f"{folders_url}?name=filter") as resp:
+        ans = await resp.json()
+        assert resp.status == 200, f"HTTP Status code error {resp.status} {ans}"
```

Comment:
i think we need to assert status 200 before the reading the response, and i see this is used differently below, sometimes we check status first then the response, sometimes the other way around

Reply:
this is done because tests were failing and needed to get an error message. Without returning the response message there was no real explanation for what went wrong. This actually works even it's not consistent with other tests.  I thought to leave it like that and maybe continue doing so?

Reply:
then good, let us leave it like that.